godotenv.go

```diff
@@ -258,30 +307,49 @@ func parseLine(line string, envMap map[string]string) (key string, value string,
 		splitString = strings.SplitN(line, ":", 2)
 	}
 
-	if len(splitString) != 2 {
-		err = errors.New("Can't separate key from value")
+	// Parse the key
+	key = strings.TrimSpace(strings.TrimPrefix(splitString[0], "export "))
+	key = exportRegex.ReplaceAllString(key, "$1")
+	if err = validateVariableName(key); err != nil {
 		return
 	}
 
-	// Parse the key
-	key = splitString[0]
-	if strings.HasPrefix(key, "export") {
-		key = strings.TrimPrefix(key, "export")
+	// Environment inherited variable
+	if firstEquals < 0 && firstColon < 0 {
+		value = ""
+		v, ok := lookupFn(strings.TrimSpace(key))
+		if ok {
+			value = v
+		}
+		return
 	}
-	key = strings.TrimSpace(key)
 
-	key = exportRegex.ReplaceAllString(splitString[0], "$1")
+	if len(splitString) != 2 {
+		err = errors.New("Can't separate key from value")
+		return
+	}
 
 	// Parse the value
 	value = parseValue(splitString[1], envMap)
 	return
 }
 
+func validateVariableName(key string) error {
```

Comment:
Did we actually need a validation for this? Note that there's no strict rules for variable names; this regex would also exclude Windows `%VARIABLE%`.

We made this mistake in the Docker Engine, and had to revert; https://github.com/moby/moby/pull/16608

Generally, I would consider the variable name to be:

- "anything" before the `=` (if present)
- except when starting with `#` (comment)
- and white space stripped

To be a "valid" variable.

Code that consumes this library can decide what to do with those (if anything); eg for docker we pass any variable to the runtime, and have the kernel/shell decide if it's accepted.